qiskit/extensions/quantum_initializer/state_preparation.py

```diff
@@ -0,0 +1,491 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Prepare a quantum state from the state where all qubits are 0."""
+
+import math
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.library.standard_gates.x import CXGate, XGate
+from qiskit.circuit.library.standard_gates.h import HGate
+from qiskit.circuit.library.standard_gates.s import SGate, SdgGate
+from qiskit.circuit.library.standard_gates.ry import RYGate
+from qiskit.circuit.library.standard_gates.rz import RZGate
+from qiskit.circuit.exceptions import CircuitError
+
+_EPS = 1e-10  # global variable used to chop very small numbers to zero
+
+
+class StatePreparation(Gate):
+    """Complex amplitude state preparation.
+
+    Class that implements the (complex amplitude) state preparation of some
+    flexible collection of qubit registers.
+    """
+
+    def __init__(self, params, num_qubits=None, inverse: bool = False):
```

Comment:
Would it be better to set the label default to `"State Preparation"` or to just update the name attribute on line 78? I'd need to add more logic for the inverse naming if we want to set default labels

Reply:
oh if I set the name to `"State Preparation"` it draws as `State preparation` that's annoying 🤔

Reply:
Yeah, but if you set the `label` to "State Preparation" I think it should draw as such. I think if label is None, then it will default to `name.capitalize()`.